File: benchmark/src/python/higher_order.py
import time
from functools import reduce

base = list(range(1000))
R = 2000
acc = 0
start = time.time() * 1000
for _ in range(R):
    d = list(map(lambda x: x * 2 + 1, base))
    f = list(filter(lambda x: x % 3 != 0, d))
    s = reduce(lambda a, x: a + x, f, 0)
    acc = (acc + s + len(f)) % 1000003
end = time.time() * 1000

print(f"Result: {acc}")
print(f"Time: {int(end - start)}ms")

File: benchmark/src/ruby/higher_order.rb
base = (0...1000).to_a
R = 2000
acc = 0
start = Time.now.to_f * 1000
R.times do
    d = base.map { |x| x * 2 + 1 }
    f = d.select { |x| x % 3 != 0 }
    s = f.reduce(0) { |a, x| a + x }
    acc = (acc + s + f.length) % 1000003
end
finish = Time.now.to_f * 1000
puts "Result: #{acc}"
puts "Time: #{(finish - start).round}ms"
